### PR TITLE
Using RLTest for the flow tests. Enabled valgrind checks on CI. Fix small leak on TopK RDB load

### LINKED ISSUE #272: Add valgrind to circleCI

### PR DESCRIPTION
**Pull request by filipecosta90**
The following PR:
- organizes the tests in the same manner as the other modules ( tests/unit and tests/flow ). 
- moves from python2 to python3 on the flow tests
- moves from RMTest to RLTest, and enables testing with replication, aof, cluster ( single shard to avoid cross slot issues on the tests for now )
- enables testing for leaks both locally and on CI ( example `make SIMPLE=1 VGD=1 test`
- fixes #272 
- It's important to mention that it reduces coverage due to config test removal. Testing for the load time config variations needs to be added ( either now or in the future.. )
- @ashtul some previous test portions were specific for AOF, rdb reload, etc... Given we have that covered by the --aof test variation I've commented those portions out with `TODO: re-enable this portion after RLTest migration`

### CODE REVIEW

File: tests/flow/test_cms.py [hunk 1]
```diff
@@ -1,99 +1,114 @@
...
+from RLTest import Env
+from redis import ResponseError
+
+is_valgrind = True if ('VGD' in os.environ or 'VALGRIND' in os.environ) else False
+
+
+class testCMS():
+    def __init__(self):
+        self.env = Env(decodeResponses=True)
+        self.assertOk = self.env.assertTrue
```
Comment (filipecosta90) [inline tests/flow/test_cms.py:14]: lets remove it prior merging...

File: tests/flow/test_cuckoo.py [hunk 2]
```diff
@@ -90,26 +111,29 @@ def test_scandump(self):
...
-        self.cmd('del', 'cf')
-        self.assertRaises(ResponseError, self.cmd, 'cf.loadchunk', 'cf')
-        self.assertRaises(ResponseError, self.cmd, 'cf.loadchunk', 'cf', 'str')
-        for chunk in chunks:
-            print("Loading chunk... (P={}. Len={})".format(chunk[0], len(chunk[1])))
-            self.cmd('cf.loadchunk', 'cf', *chunk)
-
-        for x in xrange(maxrange):
-            self.assertEqual(1, self.cmd('cf.exists', 'cf', str(x)))
+        # TODO: re-enable this portion after RLTest migration
```
Comment (filipecosta90) [inline tests/flow/test_cuckoo.py:114]: keep this

**Comment by rafie** [comment]
@filipecosta90 LGTM. I'll approve it a la carte, then iterate on it a little in subsequent PRs.

**Review by rafie — APPROVED** [review]

**Comment by filipecosta90** [comment]
Thank you @rafie. RLTest will definitely help a lot!

**Pull request merged**